optimizely/odp/odp_config.py

```diff
@@ -0,0 +1,96 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Optional
+from threading import Lock
+
+
+class OdpConfig:
+    """
+    Contains configuration used for ODP integration.
+
+    Args:
+        api_host: The host URL for the ODP audience segments API (optional).
+        api_key: The public API key for the ODP account from which the audience segments will be fetched (optional).
+        segments_to_check: An array of all ODP segments used in the current datafile
+                            (associated with api_host/api_key).
+    """
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        api_host: Optional[str] = None,
+        segments_to_check: Optional[list[str]] = None
+    ) -> None:
+        self.api_key = api_key
+        self.api_host = api_host
+        self.segments_to_check = segments_to_check or []
+        self.lock = Lock()
+
+    def update(self, api_key: Optional[str], api_host: Optional[str], segments_to_check: list[str]) -> bool:
+        """
+        Override the ODP configuration.
+
+        Args:
+            api_host: The host URL for the ODP audience segments API (optional).
+            api_key: The public API key for the ODP account from which the audience segments will be fetched (optional).
+            segments_to_check: An array of all ODP segments used in the current datafile
+                               (associated with api_host/api_key).
+
+        Returns:
+            True if the provided values were different than the existing values.
+        """
+        updated = False
+        with self.lock:
+            if self.api_key != api_key or self.api_host != api_host or self.segments_to_check != segments_to_check:
+                self.api_key = api_key
+                self.api_host = api_host
+                self.segments_to_check = segments_to_check
+                updated = True
+
+        return updated
+
+    def get_api_host(self) -> Optional[str]:
+        with self.lock:
+            return self.api_host
+
+    def set_api_host(self, api_host: Optional[str]) -> None:
+        with self.lock:
+            self.api_host = api_host
+
+    def get_api_key(self) -> Optional[str]:
+        with self.lock:
+            return self.api_key
+
+    def set_api_key(self, api_key: Optional[str]) -> None:
+        with self.lock:
+            self.api_key = api_key
+
+    def get_segments_to_check(self) -> Optional[list[str]]:
+        with self.lock:
+            return self.segments_to_check.copy()
+
+    def set_segments_to_check(self, segments_to_check: list[str]) -> None:
+        """
+        Replace qualified segments with provided list of segments
```

Comment:
Minor...I think we could use proper sentence with period etc at the end. The rest os SDK does that. Maybe for consistency? Then we can discuss if generally we care about periods or not? (goes for all comments in the file)